packages/core/src/components/overlay2/overlay2.tsx

```diff
@@ -55,6 +55,9 @@ export interface OverlayInstance {
     /** Document "focus" event handler which needs to be attached & detached appropriately. */
     handleDocumentFocus: (e: FocusEvent) => void;
 
+    /** Document "mousedown" event handler which needs to be attached & detached appropriately. */
```

Comment:
now that there's an appropriate global context manager to handle this kind of logic, we should consider refactoring this behavior to have a single handler for the whole overlay stack instead of separate handlers for each overlay which need to be attached/detached. that would likely be safer and less prone to memory leaks.